runtime/src/vm/opcode.rs

```diff
@@ -3074,110 +2560,188 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
-          } else {
-            break
-          }
-        }
-        let result = if is_ok { 1i64 } else { 0i64 };
-        let mut hand_mem = mem.write().await;
-        hand_mem.new_fractal(args[2]);
-        hand_mem.push_fractal_fixed(args[2], result);
-        hand_mem.push_nested_fractal_mem(args[2], out);
-        drop(hand_mem); // drop write lock
+      };
+      let result = if is_ok { 1i64 } else { 0i64 };
+      let mut hand_mem = mem.write().await;
+      hand_mem.new_fractal(args[2]);
+      hand_mem.push_fractal_fixed(args[2], result);
+      hand_mem.push_nested_fractal(args[2], HandlerMemory::str_to_hm(&result_str));
+      drop(hand_mem); // drop write lock
+    };
+    return Box::pin(fut);
+  });
+
+  async fn http_listener(req: Request<Body>) -> Result<Response<Body>, Infallible> {
+    // Create a new event handler memory to add to the event queue
+    let mut event = HandlerMemory::new(None, 1);
+    // Grab the URL
+    let url_str = req.uri().to_string();
+    let url = HandlerMemory::str_to_hm(&url_str);
+    // Grab the headers
+    let headers = req.headers();
+    let mut headers_hm = HandlerMemory::new(None, headers.len() as i64);
+    let mut i = 0;
+    for (key, val) in headers.iter() {
+      let key_str = key.as_str();
+      let val_str = val.to_str().unwrap();
+      let mut header_hm = HandlerMemory::new(None, 2);
+      header_hm.write_fractal(0, HandlerMemory::str_to_hm(key_str));
+      header_hm.write_fractal(1, HandlerMemory::str_to_hm(val_str));
+      headers_hm.write_fractal(i, header_hm);
+      i = i + 1;
+    }
+    // Grab the body, if any
+    let body_req = hyper::body::to_bytes(req.into_body()).await;
+    // If we error out while getting the body, just close this listener out immediately
+    if body_req.is_err() {
+      return Ok(Response::new("Conection terminated".into()));
+    }
+    let body_str = str::from_utf8(&body_req.unwrap()).unwrap().to_string();
+    let body = HandlerMemory::str_to_hm(&body_str);
+    // Generate a connection ID
+    let conn_id = OsRng.next_u64() as i64;
+    // Populate the event and emit it
+    event.new_fractal(0);
+    event.push_nested_fractal(0, url);
+    event.push_nested_fractal(0, headers_hm);
+    event.push_nested_fractal(0, body);
+    event.push_fractal_fixed(0, conn_id);
+    let event_emit = EventEmit {
+      id: i64::from(BuiltInEvents::HTTPCONN),
+      payload: Some(event),
+    };
+    let event_tx = EVENT_TX.get().unwrap();
+    if event_tx.send(event_emit).is_err() {
+      eprintln!("Event transmission error");
+      std::process::exit(3);
+    }
+    // Get the HTTP responses lock and periodically poll it for responses from the user code
+    // TODO: Swap from a timing-based poll to getting the waker to the user code so it can be
+    // woken only once and reduce pressure on this lock.
+    let responses = Arc::clone(&HTTP_RESPONSES);
+    let response_hm = poll_fn(|cx: &mut Context<'_>| -> Poll<HandlerMemory> {
```

Comment:
oh rust..

Reply:
I am glad I found that thing, though! I wasn't sure how I was going to implement this without something of this nature.

And I should be able to make it more efficient (and not really a poll at all) in the future.

We could also have timeout support at some point to make sure that there aren't dead connections piling up.